src/main/java/com/redhat/exhort/utils/PythonControllerBase.java

```diff
@@ -304,96 +278,154 @@ private List<Map<String, Object>> getDependenciesImpl(
-    return getStringValueEnvironment(EnvVar, "").trim().equals("")
-        ? Operations.runProcessGetOutput(pythonEnvironmentDir, cmdList)
-        : new String(Base64.getDecoder().decode(getStringValueEnvironment(EnvVar, "")));
+  List<PythonDependency> getDependencyTreeJsonFromPipDepTree() {
+    executeCommandOrExtractFromEnv(
+        "EXHORT_PIP_PIPDEPTREE", pipBinaryLocation, "install", "pipdeptree");
+
+    String pipdeptreeJsonString;
+    if (isVirtualEnv()) {
+      pipdeptreeJsonString =
+          executeCommandOrExtractFromEnv("EXHORT_PIP_PIPDEPTREE", "./bin/pipdeptree", "--json");
+    } else if (isRealEnv()) {
+      pipdeptreeJsonString =
+          executeCommandOrExtractFromEnv(
+              "EXHORT_PIP_PIPDEPTREE", pathToPythonBin, "-m", "pipdeptree", "--json");
+    } else {
+      pipdeptreeJsonString =
+          executeCommandOrExtractFromEnv(
+              "EXHORT_PIP_PIPDEPTREE", "./bin/pipdeptree", "--json", "--python", pathToPythonBin);
+    }
+    if (debugLoggingIsNeeded()) {
+      String pipdeptreeMessage =
+          String.format(
+              "Package Manager pipdeptree --json command result output -> %s %s",
+              System.lineSeparator(), pipdeptreeJsonString);
+      log.info(pipdeptreeMessage);
+    }
+    return mapToPythonDependencies(pipdeptreeJsonString);
+  }
+
+  public static List<PythonDependency> mapToPythonDependencies(String jsonString) {
+    // Parse JSON string using ObjectMapper
+    ObjectMapper mapper = new ObjectMapper();
+    JsonNode rootNode = null;
+    try {
+      rootNode = mapper.readTree(jsonString);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+    List<PythonDependency> dependencies = new ArrayList<>();
+
+    // Check if it's an array
+    if (rootNode.isArray()) {
+      for (JsonNode dependencyNode : rootNode) {
+        if (dependencyNode.isObject()) {
+          ObjectNode dependencyObject = (ObjectNode) dependencyNode;
+
+          // Extract information from the nested "package" object
+          JsonNode packageNode = dependencyObject.get("package");
+          String name = packageNode.get("package_name").asText();
+          String version = packageNode.get("installed_version").asText();
+
+          // Extract dependencies (might be an array or an empty object)
+          JsonNode dependenciesElement = dependencyObject.get("dependencies");
+          List<String> depList = new ArrayList<>();
+          if (dependenciesElement.isArray()) {
+            // Loop through the dependencies array and add names
+            for (JsonNode depNode : dependenciesElement) {
+              String depName = depNode.get("package_name").asText();
+              depList.add(depName);
+            }
+          }
+
+          // Create a PythonDependency object and add it to the list
+          PythonDependency dependency = new PythonDependency(name, version, depList);
+          dependencies.add(dependency);
+        }
```

Comment:
@JudeNiroshan very good that you decided to use jackson jsonNode in order to parse the json from pipdeptree
it's not mandatory, but i'm recommending to switch the old for loops iterating over collections , with streams.

Reply:
6e6af49513231ceb8f15b031fcc3535fe724886a